### PR TITLE
change: move config_center under deployment and named as config_provider

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/stand-alone.md [hunk 1]
```diff
@@ -34,13 +34,15 @@ The routing rules in the `conf/apisix.yaml` file are loaded into memory immediat
 
 Since the current Admin API is based on the etcd configuration center solution, enable Admin API is not allowed when the Stand-alone mode is enabled.
 
-To enable Stand-alone mode, we can set `apisix.config_center` to `yaml` and disable Admin API in file `conf/config.yaml`.
+The Stand-alone mode can only be enabled when set the role of APISIX as data plane. We can set `deployment.role` to `data_plane` and `deployment.role_data_plane.config_provider` to `yaml`, and also need to disable Admin API in file `conf/config.yaml`.
```
Comment (spacewander) [inline docs/en/latest/stand-alone.md:37]: When we set `deployment.role` to `data_plane`, the admin API is disabled automatically.

Comment (tzssangglass) [reply docs/en/latest/stand-alone.md]: updated.

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Comment by sober-wang** [comment]
this pr is merged, but apisix repo version is v2.15.1 . 

图片

can you tell me how to use  the `apisix.config_center: yaml` config . 
Just learning apisix , and create a etcd 3 node caluster  , it looks a little complicated.

**Comment by tzssangglass** [comment]
> 此 pr 已合并，但 apisix repo 版本为 v2.15.1

v2.15.1  no need to be concerned about changes to this PR.

**Pull request merged**